src/fcntl.rs

```diff
@@ -549,6 +558,20 @@ pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
             F_GETPIPE_SZ => libc::fcntl(fd, libc::F_GETPIPE_SZ),
             #[cfg(any(target_os = "linux", target_os = "android"))]
             F_SETPIPE_SZ(size) => libc::fcntl(fd, libc::F_SETPIPE_SZ, size),
+            #[cfg(any(target_os = "dragonfly", target_os = "netbsd", target_os = "macos", target_os = "ios"))]
+            F_GETPATH(path) => {
+                let mut buffer = vec![0; libc::PATH_MAX as usize];
+                res = libc::fcntl(fd, libc::F_GETPATH, buffer.as_ptr());
+                if res < 0 {
+                    return Errno::result(res);
+                } else {
+                    let len = buffer.iter().position(|b| *b == 0).unwrap();
+                    buffer.truncate(len as usize);
+                    buffer.shrink_to_fit();
+                    *path = PathBuf::from(OsString::from_vec(buffer));
+                    return Errno::result(res);
+                }
+            },
```

Comment:
It might be better to write it like this:

```rust
let mut buffer = vec![0; libc::PATH_MAX as usize];
let res = libc::fcntl(fd, libc::F_GETPATH, buffer.as_ptr());
let ok_res = Errno::result(res)?
let len = buffer.iter().position(|b| *b == 0).unwrap();
buffer.truncate(len as usize);
buffer.shrink_to_fit();
*path = PathBuf::from(OsString::from_vec(buffer));

ok_res
```